src/mgconfitem.cpp

```diff
@@ -84,7 +89,7 @@ void MGConfItem::update_value(bool emit_signal)
     if (new_value != priv->value || new_value.userType() != priv->value.userType()) {
         priv->value = new_value;
         if (emit_signal)
-            emit valueChanged();
+            QMetaObject::invokeMethod(this, "emitValueChanged", Qt::QueuedConnection);
```

Comment:
I'm pretty sure you can just pass the signal to invokeMethod with the same effect.